syntax = "proto3";

import "google/protobuf/any.proto";
import "google/protobuf/descriptor.proto";

message EngineSpecific {
    string msg_name = 1;
    google.protobuf.FileDescriptorProto msg_proto = 2;
    google.protobuf.Any msg_content = 3;
}

message State {
    string basic_parameter = 1;
    EngineSpecific engine_specific = 2;
}
